server/src/main/java/org/apache/druid/guice/DruidProcessingModule.java

```diff
@@ -135,6 +136,26 @@ public ExecutorService getProcessingExecutorService(
     );
   }
 
+  @Provides
+  @ManageLifecycle
+  public LifecycleForkJoinPoolProvider getMergeProcessingPoolProvider(DruidProcessingConfig config)
+  {
+    return new LifecycleForkJoinPoolProvider(
+        config.getNumThreadsMergePool(),
+        ForkJoinPool.defaultForkJoinWorkerThreadFactory, // todo: ?
```

Comment:
I guess this comment is related to a custom prioritized factory? Please update the comment.

Reply:
Ah, no this was for whether or not I was going to set custom thread names as @drcrallen did in #5913, see here https://github.com/apache/incubator-druid/pull/5913/files#diff-fa77f3240f159148c9eac7faa94c516fR47. I've decided against it for now and removed the todo since all of the logs are debug level other than the unhandled exception handler I just added, but am open to discussion.

Reply:
Maybe still it could be worth to be more debug friendly for who are not familiar with Druid internal.